File: dingco-python/2nd_week/2_extra_josephus.py
def josephus(n, k):
    people = list(range(1, n+1))

    remove_order = []
    next_target = k-1

    while people:
        popped = people.pop(next_target)
        remove_order.append(popped)

        if people:
            next_target = (next_target + (k-1)) % len(people)

    ret = "<" + ", ".join(map(str, remove_order)) + ">"
    print(ret)

n, k = map(int, input().split())
josephus(n, k)

File: dingco-python/2nd_week/2_extra_josephus.swift
func josephus(_ n: Int, _ k: Int) -> [Int] {
    var removeOrder = [Int]()

    var people = [Int]()
    for i in 1...n {
        people.append(i)
    }
    var nextPopIndex = k - 1

    while !people.isEmpty {
        let popped = people.remove(at: nextPopIndex)
        removeOrder.append(popped)

        if !people.isEmpty {
            nextPopIndex = (nextPopIndex + (k - 1)) % people.count
        }
    }

    return removeOrder
}

let inputs = readLine()!.split(separator:" ").map { Int($0)! }
let result = josephus(inputs[0], inputs[1])
print("<\(result.map { "\($0)" }.joined(separator: ", "))>")
